json_test.go

```diff
@@ -41,3 +43,109 @@ func TestJsonUnmarshalQuery(t *testing.T) {
 		return
 	}
 }
+
+type JsonValues struct {
+	Text      string      `json:"text"`
+	Integer   int         `json:"integer"`
+	Decimal   float64     `json:"decimal"`
+	Bool      bool        `json:"bool"`
+	Array     []string    `json:"array"`
+	SubObject *JsonValues `json:"obj"`
+}
+
+func TestJsonUnmarshalError(t *testing.T) {
+	var (
+		target JsonValues
+		jue    JsonUnmarshalError
+	)
+
+	type testCase struct {
+		rawJson  string
+		source   string
+		target   string
+		property string
+	}
+	for idx, c := range []testCase{
+		{
+			rawJson:  `{"text": false}`,
+			source:   "bool",
+			target:   "string",
+			property: "JsonValues.text",
+		},
+		{
+			rawJson:  `{"text": 123}`,
+			source:   "number",
+			target:   "string",
+			property: "JsonValues.text",
+		},
+		{
+			rawJson:  `{"text": 123.456}`,
+			source:   "number",
+			target:   "string",
+			property: "JsonValues.text",
+		},
+		{
+			rawJson:  `{"integer": "invalid"}`,
+			source:   "string",
+			target:   "int",
+			property: "JsonValues.integer",
+		},
+		{
+			rawJson:  `{"decimal": "invalid"}`,
+			source:   "string",
+			target:   "float64",
+			property: "JsonValues.decimal",
+		},
+		{
+			rawJson:  `{"bool": 123}`,
+			source:   "number",
+			target:   "bool",
+			property: "JsonValues.bool",
+		},
+		{
+			rawJson:  `{"array": false}`,
+			source:   "bool",
+			target:   "[]string",
+			property: "JsonValues.array",
+		},
+		{
+			rawJson:  `{"array": [false, false]}`,
+			source:   "bool",
+			target:   "string",
+			property: "JsonValues.array",
+		},
+		{
+			rawJson:  `{"obj": false}`,
+			source:   "bool",
+			target:   "golib.JsonValues",
+			property: "JsonValues.obj",
+		},
+		{
+			rawJson:  `{"obj": {"array": "invalid"}}`,
+			source:   "string",
+			target:   "[]string",
+			property: "JsonValues.obj.array",
```

Comment:
Diese vielen Tests sind im Grund nicht so relevant, weil sie ja alle denselben Code Path in der golib und auch im JSON stdlib code testen.

Was wichtiger wäre ist ein Test der die non struct error message (3 value message) testet, das habe ich nicht gesehen. Und das berührt dann wiederum mehr Code.